src/main/java/poolsheen/Ui.java

```diff
@@ -0,0 +1,143 @@
+package poolsheen;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import poolsheen.task.Task;
+
+/**
+ * Represents everything that involves interactions with the user.
+ */
+public class Ui {
+    /** The indentation for all printed Poolsheen lines. */
+    public static final String BEGIN_SPACE = "      ";
+
+    /** The Poolsheen logo. */
+    private static final String LOGO = "\n" + "POOLSHEEN";
+
+    /** The message first printed when the Poolsheen program runs. */
+    private static final String WELCOME_MESSAGE = "Hello from" + LOGO + "\n" +
+            "Type something in for Poolsheen to respond to you:";
+
+    /** The message printed last whenever Poolsheen replys. */
+    private static final String LAST_REPLY = "meow *_*";
+
+    /** A horiontal line to distinguish each interaction the user has with Poolsheen. */
+    private static final String HORIZONTAL_LINE = "---------------";
+
+    /** The message last printed when the Poolsheen program ends. */
+    private static final String GOODBYE_MESSAGE = "MeoAww... See you next time :(\n"
+            + "THE POOLSHEEN PROGRAM HAS STOPPED RUNNING";
+
+    /** The Scanner object which the Poolsheen program uses to interact with the user. */
+    private Scanner s;
+
+    /**
+     * A public constructor to initialise a Ui object.
+     */
+    public Ui() {
+        s = new Scanner(System.in);
+        System.out.println("Poolsheen UI has loaded");
+    }
+
+    /**
+     * Prints a message whenever Poolsheen encounters a loading error.
+     */
+    public static void showLoadingError() {
+        System.out.println("Poolsheen has encountered a loading error");
+    }
+
+    /**
+     * Prints a formatted message by Poolsheen.
+     *
+     * @param message The message to be printed.
+     */
+    public void say(String message) {
+        System.out.println(HORIZONTAL_LINE + "\n" +
+                newLine(message) +
+                newLine(LAST_REPLY) + HORIZONTAL_LINE);
+    }
+
+    /**
+     * Formats and prints a UI Error to the user.
+     *
+     * @param errMsg The string obtained from an error's getMessage function.
+     */
+    public void showError(String errMsg, String errorType) {
+        System.out.println(HORIZONTAL_LINE + "\n" +
+                newLine("The following UI error has occurred") +
+                newLine("The error type is: " + errorType) +
+                newLine(errMsg) +
+                newLine("Please try again!") +
+                newLine(LAST_REPLY) + HORIZONTAL_LINE);
+    }
```

Comment:
The abstraction of constant strings pays off well here! With the `newLine` function the code looks clean and readable, contrast to having a slew of `System.out.println` messages. Well done!